Engine_Revit_UI/Query/ElementIds/ElementIdsOfElementsOnly.cs

```diff
@@ -43,13 +34,22 @@ public static partial class Query
         /****              Public methods               ****/
         /***************************************************/
 
-        public static IEnumerable<ElementId> ElementIdsViewTemplate(this Document document, IEnumerable<ElementId> ids = null)
+        [Description("Get all instances of Elements as ElementIds, implies they exist in a XYZ context")]
+        [Input("document", "Revit Document where ElementIds are collected")]        
+        [Input("ids", "Optional, allows the filter to narrow the search from an existing enumerator")]
+        [Output("elementIdsOfElementsOnly", "An enumerator for easy iteration of ElementIds collected")]
+        public static IEnumerable<ElementId> ElementIdsOfElementsOnly(this Document document, IEnumerable<ElementId> ids = null)
         {
+            if (document == null)
+                return null;
+
             if (ids != null && ids.Count() == 0)
                 return new List<ElementId>();
 
             FilteredElementCollector collector = ids == null ? new FilteredElementCollector(document) : new FilteredElementCollector(document, ids.ToList());
-            return collector.OfClass(typeof(View)).Cast<View>().Where(x => x.IsTemplate).Select(x => x.Id);
+            collector.WhereElementIsNotElementType().Where(x => x.Location != null);
```

Comment:
As discussed, this line does nothing since `WhereElementIsNotElementType` method returns a new object of type `FilteredElementCollector` - could you please make sure that this error does not appear anywhere else in this PR?